embeddings_utils.go

```diff
@@ -0,0 +1,11 @@
+package openai
+
+// DotProduct Calculate dot product of two vectors.
+func DotProduct(v1, v2 []float32) float32 {
```

Comment:
Why not put it as an Embedding method?

Reply:
![image](https://github.com/sashabaranov/go-openai/assets/7778936/54516416-5934-4af6-b18d-47c4bae9af92)

@sashabaranov  like you say.

Reply:
Totally, but we don't really have `[]float32` vectors in the library except for `Embedding` struct. Might make sense to add it as `func (e Embedding) DotProduct(another Embedding)`